gravitee-apim-console-webui/src/management/settings/categories/category/category.component.ts

```diff
@@ -13,32 +13,260 @@
+          page: new FormControl<string>(category.page),
+          picture: new FormControl<unknown[]>(category.picture_url ? [category.picture_url] : []),
+          background: new FormControl<unknown[]>(category.background_url ? [category.background_url] : []),
+          highlightApi: this.highlightApiControl,
+        });
+        this.categoryDetailsInitialValue = this.categoryDetails.getRawValue();
+        this.handleReadOnly();
+      }),
+    );
+
+    this.apis$ = this.category.pipe(
+      filter((category) => !isEmpty(category)),
+      switchMap((category) => this.apiService.getAll({ category: category.key })),
+      map((apis) => apis.map((api) => ({ id: api.id, name: api.name, version: api.version, contextPath: api.context_path }))),
+    );
+
+    this.pages$ = this.refreshData.pipe(switchMap((_) => this.documentationService.listPortalPages(PageType.MARKDOWN, true)));
+  }
+
+  onSubmit() {
+    of(this.mode)
+      .pipe(
+        switchMap((mode) => (mode === 'edit' ? this.updateCategory$() : this.createCategory$())),
+        takeUntilDestroyed(this.destroyRef),
+      )
+      .subscribe({
+        next: (category) => {
+          this.snackBarService.success(`Category [${category.name}] successfully ${this.mode === 'new' ? 'created' : 'updated'}.`);
+          if (this.mode === 'new') {
+            this.router.navigate(['..', category.id], { relativeTo: this.activatedRoute });
+          } else {
+            this.refreshData.next(1);
+          }
+        },
+        error: ({ error }) => this.snackBarService.error(error.message),
+      });
+  }
+
+  private updateCategory$(): Observable<Category> {
+    return this.categoryService.get(this.category.getValue().id).pipe(
+      switchMap((category) => {
+        const newValues = this.categoryDetails.getRawValue();
+        const categoryToUpdate: UpdateCategory = {
+          ...category,
+          name: newValues.name,
+          description: newValues.description,
+          page: newValues.page,
+          hidden: newValues.hidden,
+          highlightApi: newValues.highlightApi,
+        };
+
+        if (newValues.picture.length && newValues.picture[0] instanceof NewFile) {
+          categoryToUpdate.picture = (newValues.picture[0] as NewFile).dataUrl;
+          categoryToUpdate.picture_url = undefined;
+        } else if (newValues.picture.length === 0 && !!category.picture_url) {
+          categoryToUpdate.picture_url = undefined;
+        }
+
+        if (newValues.background.length && newValues.background[0] instanceof NewFile) {
+          categoryToUpdate.background = (newValues.background[0] as NewFile).dataUrl;
+          categoryToUpdate.background_url = undefined;
+        } else if (newValues.background.length === 0 && !!category.background_url) {
+          categoryToUpdate.background_url = undefined;
+        }
+
+        return this.categoryService.update(categoryToUpdate);
```

Comment:
Should we have error handling in here?

Reply:
The error handling is in the subscribe of the onSubmit function: We just show the error snackbar with the error message.